packages/react-components/src/components/VideoGallery.tsx

```diff
@@ -426,72 +427,82 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
-          localScreenShareStreamComponent
-        ) : (
-          <GridLayout key="grid-layout" styles={styles?.gridLayout}>
-            {gridTiles}
-          </GridLayout>
-        )}
-        {horizontalGalleryPresent && (
-          <div style={{ paddingTop: '0.5rem' }}>
-            <ResponsiveHorizontalGallery
-              key="responsive-horizontal-gallery"
-              containerStyles={horizontalGalleryContainerStyle(shouldFloatLocalVideo, isNarrow)}
-              horizontalGalleryStyles={concatStyleSets(horizontalGalleryStyle(isNarrow), styles?.horizontalGallery)}
-              childWidthRem={
-                isNarrow ? SMALL_HORIZONTAL_GALLERY_TILE_SIZE_REM.width : LARGE_HORIZONTAL_GALLERY_TILE_SIZE_REM.width
-              }
-              buttonWidthRem={HORIZONTAL_GALLERY_BUTTON_WIDTH}
-              gapWidthRem={HORIZONTAL_GALLERY_GAP}
+  if (layout === 'floatingLocalVideo') {
+    return (
+      <div
+        data-ui-id={ids.videoGallery}
+        ref={containerRef}
+        className={mergeStyles(videoGalleryOuterDivStyle, styles?.root)}
+      >
+        {shouldFloatLocalVideo &&
+          !shouldFloatNonDraggableLocalVideo &&
+          localVideoTile &&
+          (horizontalGalleryPresent ? (
+            <Stack className={mergeStyles(localVideoTileContainerStyle(theme, isNarrow))}>{localVideoTile}</Stack>
+          ) : (
+            <_ModalClone
+              isOpen={true}
+              isModeless={true}
+              dragOptions={DRAG_OPTIONS}
+              styles={floatingLocalVideoModalStyle(theme, isNarrow)}
+              layerProps={{ hostId: layerHostId }}
+              maxDragPosition={modalMaxDragPosition}
+              minDragPosition={modalMinDragPosition}
+            >
+              {localVideoTile}
+            </_ModalClone>
+          ))}
+        {
+          // When we use showCameraSwitcherInLocalPreview it disables dragging to allow keyboard navigation.
+          shouldFloatNonDraggableLocalVideo && localVideoTile && remoteParticipants.length > 0 && (
+            <Stack
+              className={mergeStyles(localVideoTileWithControlsContainerStyle(theme, isNarrow), {
+                boxShadow: theme.effects.elevation8,
+                zIndex: LOCAL_VIDEO_TILE_ZINDEX
+              })}
             >
-              {horizontalGalleryTiles}
-            </ResponsiveHorizontalGallery>
-          </div>
-        )}
+              {localVideoTile}
+            </Stack>
+          )
+        }
+        <Stack horizontal={false} styles={videoGalleryContainerStyle}>
+          {screenShareParticipant ? (
+            remoteScreenShareComponent
+          ) : localParticipant?.isScreenSharingOn ? (
+            localScreenShareStreamComponent
+          ) : (
+            <GridLayout key="grid-layout" styles={styles?.gridLayout}>
+              {gridTiles}
+            </GridLayout>
+          )}
+          {horizontalGalleryPresent && (
+            <div style={{ paddingTop: '0.5rem' }}>
```

Comment:
nit:
As far as I can tell this is the exact same code as the defaultLayout.
I like the separation! And I like `ResponsiveHorizontalGallery` doesn't have these defaults and is kept my agnostic. But consider in `ResponsiveHorizontalGallery.tsx` adding a 
```tsx
/**
 * A ResponsiveHorizontalGallery styled for the @link{VideoGallery}
 */
const VideoGalleryResponsiveHorizontalGallery = (props): JSX.Element => (
   <ResponsiveHorizontalGallery
              key="responsive-horizontal-gallery"
              containerStyles={horizontalGalleryContainerStyle(false, isNarrow)}
              horizontalGalleryStyles={concatStyleSets(horizontalGalleryStyle(isNarrow), styles?.horizontalGallery)}
              childWidthRem={
                isNarrow ? SMALL_HORIZONTAL_GALLERY_TILE_SIZE_REM.width : LARGE_HORIZONTAL_GALLERY_TILE_SIZE_REM.width
              }
              buttonWidthRem={HORIZONTAL_GALLERY_BUTTON_WIDTH}
              gapWidthRem={HORIZONTAL_GALLERY_GAP}
            >
              {horizontalGalleryElements}
   </ResponsiveHorizontalGallery>
);
```